src/rest_framework_dso/filters.py

```diff
@@ -73,6 +75,32 @@ def __init__(self, field_name=None, lookup_expr="exact", **kwargs):
         super().__init__(field_name, lookup_expr, **kwargs)
 
 
+class EffectiveFilter(filters.CharFilter):
+    """Filter by effective date."""
+
+    filter_name = "inWerkingOp"
+    label = "Filter values effective on provided date/time."
+
+    def __init__(self, start_field, end_field, lookup_expr="exact", **kwargs):
+        self.start_field = self.convert_field_name(start_field)
+        self.end_field = self.convert_field_name(end_field)
+        super().__init__(field_name=self.start_field, lookup_expr=lookup_expr, **kwargs)
+
+    def filter(self, qs, value):
+        if value.strip() == "":
+            return qs
+        return qs.filter(
+            **{f"{self.start_field}__lte": value, f"{self.end_field}__gt": value}
+        )
+
+    def convert_field_name(self, field_name):
+        if "." in field_name:
+            return "__".join(
+                [self.convert_field_name(part) for part in field_name.split(".")]
+            )
+        return slugify(field_name, sign="_")
```

Comment:
Let op, 'sign' wordt niet meer ondersteund in nieuwe versie van string-utils. Is nu 'separator'.